src/System.Net.Http/tests/FunctionalTests/LoopbackServer.cs

```diff
@@ -0,0 +1,112 @@
+                using (var reader = new StreamReader(stream, Encoding.ASCII))
+                {
+                    // Read past request headers.
+                    string line;
+                    while (!string.IsNullOrEmpty(line = reader.ReadLine())) ;
+
+                    // Determine response transfer headers.
+                    string transferHeader = null;
+                    string content = "This is some response content.";
+                    if (transferType == TransferType.ContentLength)
+                    {
+                        if (transferError == TransferError.ContentLengthTooLarge)
+                        {
+                            transferHeader = $"Content-Length: {content.Length + 42}\r\n";
+                        }
+                        else
+                        {
+                            transferHeader = $"Content-Length: {content.Length}\r\n";
+                        }
+                    }
+                    else if (transferType == TransferType.Chunked)
+                    {
+                        transferHeader = "Transfer-Encoding: chunked\r\n";
+                    }
+
+                    // Write response.
+                    using (var writer = new StreamWriter(stream, Encoding.ASCII))
+                    {
+                        writer.Write("HTTP/1.1 200 OK\r\n");
+                        writer.Write($"Date: {DateTimeOffset.UtcNow:R}\r\n");
+                        writer.Write("Content-Type: text/plain\r\n");
+
+                        if (!string.IsNullOrEmpty(transferHeader))
+                        {
+                            writer.Write(transferHeader);
+                        }
+
+                        writer.Write("\r\n");
+                        if (transferType == TransferType.Chunked)
+                        {
+                            string chunkSizeInHex;
+                            if (transferError == TransferError.ChunkSizeTooLarge)
+                            {
+                                chunkSizeInHex = string.Format("{0:x}\r\n", content.Length + 42);
+                            }
+                            else
+                            {
+                                chunkSizeInHex = string.Format("{0:x}\r\n", content.Length);
+                            }
+                            writer.Write(chunkSizeInHex);
+                            writer.Write($"{content}\r\n");
+                            if (transferError != TransferError.MissingChunkTerminator)
+                            {
+                                writer.Write("0\r\n\r\n");
+                            }
+                        }
+                        else
+                        {
+                            writer.Write($"{content}\r\n");
+                        }
+                        writer.Flush();
+                    }
+
+                    client.Shutdown(SocketShutdown.Both);
```

Comment:
I know this is preexisting from my change, but it's there any chance anything above throws, and if so, what happens if this shutdown doesn't happen? Just wondering if this should be in a finally.